### PR TITLE
feat(delete): delete company certificates

### PR DESCRIPTION
**Pull request by AnuragNagpure**
## Description

DELETE Company Certificate.
 /api/administration/companydata/companyCertificate/document/{documentId}

## Why
While implementing the logic requirement mentioned below should be taken care.
When running the endpoint - first validate if the certificate is assigned to the same company_id as the acting user company_id

if yes, proceed (see below)
if no; stop and exception/error
Set the respective document inside the table portal.documents to "INACTIVE" and update the "last_date_changed" date as well as the "last_editor_id"

Afterwards set inside the table portal.company_certificates.status to "INACTIVE" 

## Issue

#467 

## Checklist

Please delete options that are not relevant.

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/docs/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally
- [x] I have added tests that prove my changes work
- [x] I have checked that new and existing tests pass locally with my changes
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/portalbackend/PortalBackend.DBAccess/Repositories/CompanyCertificateRepository.cs [hunk 1]
```diff
@@ -99,4 +99,31 @@ sorting switch
...
+
+    public Task<(Guid DocumentId, DocumentStatusId DocumentStatusId, Guid CompanyCertificateId, bool IsSameCompany)> GetCompanyCertificateDocumentDetailsForIdUntrackedAsync(Guid documentId, Guid companyUserId) =>
+            _context.Documents
+            .AsNoTracking()
+            .Where(x => x.Id == documentId)
+            .Select(document => ((Guid DocumentId, DocumentStatusId DocumentStatusId, Guid CompanyCertificateId, bool IsSameCompany))
+                new(document.Id,
+                    document.DocumentStatusId,
+                    document.CompanyCertificates.Select(x => x.Id).SingleOrDefault(),
+                    document.CompanyUserId == companyUserId))
```
Comment (Phil91) [inline src/portalbackend/PortalBackend.DBAccess/Repositories/CompanyCertificateRepository.cs:111]: With this check only the user who uploaded the certificate can delete it. Please change the check to the companyId of the CompanyUser and check with the companyID of the current logged in user.

**Review by Phil91 — CHANGES_REQUESTED** [review]

File: src/portalbackend/PortalBackend.DBAccess/Repositories/CompanyCertificateRepository.cs [hunk 2]
```diff
@@ -99,4 +99,31 @@ sorting switch
                 companyCertificate.ValidTill
                 ))
         .SingleOrDefaultAsync();
+
+    public Task<(Guid DocumentId, DocumentStatusId DocumentStatusId, Guid CompanyCertificateId, bool IsSameCompany)> GetCompanyCertificateDocumentDetailsForIdUntrackedAsync(Guid documentId, Guid companyUserId) =>
+            _context.Documents
+            .AsNoTracking()
+            .Where(x => x.Id == documentId)
+            .Select(document => ((Guid DocumentId, DocumentStatusId DocumentStatusId, Guid CompanyCertificateId, bool IsSameCompany))
```
Comment (Phil91) [inline src/portalbackend/PortalBackend.DBAccess/Repositories/CompanyCertificateRepository.cs:107]:
```suggestion
            .Select(document => new ValueTuple<Guid, DocumentStatusId, Guid, bool>
```

**Review by Phil91 — CHANGES_REQUESTED** [review]

File: src/portalbackend/PortalBackend.DBAccess/Repositories/CompanyCertificateRepository.cs (hunk 2 above)
Comment (AnuragNagpure) [reply src/portalbackend/PortalBackend.DBAccess/Repositories/CompanyCertificateRepository.cs]: Value tuple added as per suggestion

File: src/portalbackend/PortalBackend.DBAccess/Repositories/CompanyCertificateRepository.cs [hunk 3]
```diff
@@ -99,4 +99,31 @@ sorting switch
...
         .SingleOrDefaultAsync();
+
+    public Task<(Guid DocumentId, DocumentStatusId DocumentStatusId, Guid CompanyCertificateId, bool IsSameCompany)> GetCompanyCertificateDocumentDetailsForIdUntrackedAsync(Guid documentId, Guid companyId) =>
+            _context.Documents
+            .AsNoTracking()
+            .Where(x => x.Id == documentId)
+            .Select(document => new ValueTuple<Guid, DocumentStatusId, Guid, bool>(
+                    document.Id,
+                    document.DocumentStatusId,
+                    document.CompanyCertificates.Select(x => x.Id).SingleOrDefault(),
```
Comment (Phil91) [inline src/portalbackend/PortalBackend.DBAccess/Repositories/CompanyCertificateRepository.cs:110]: It seems like database wise it's possible to have multiple CompanyCertificates for one document... We should implement an error handling for the case we have multiple CompanyCertificates. With this logic when having multiple CompanyCertificates for one Document the user would receive an UnhandledServer Exception

**Review by Phil91 — CHANGES_REQUESTED** [review]

File: src/portalbackend/PortalBackend.DBAccess/Repositories/CompanyCertificateRepository.cs (hunk 3 above)
Comment (AnuragNagpure) [reply src/portalbackend/PortalBackend.DBAccess/Repositories/CompanyCertificateRepository.cs]: changes for checking multiple company certificates is implemented

**Review by Phil91 — DISMISSED** [review]

**Review by Phil91 — APPROVED** [review]

**Pull request merged**